core/src/main/java/org/infinispan/xsite/statetransfer/XSiteStatePushCommand.java

```diff
@@ -76,7 +76,7 @@ public void writeTo(ObjectOutput output) throws IOException {
    @Override
    public void readFrom(ObjectInput input) throws IOException, ClassNotFoundException {
       timeoutMillis = input.readLong();
-      chunk = MarshallUtil.unmarshallArray(input, XSiteState[]::new);
+      chunk = MarshallUtil.unmarshallArray(input, XSiteState::array);
```

Comment:
Here too, I think 0-length arrays are quite rare (except maybe in tests that have little to no data in each segment).